dkg/pedersen/handler.go

```diff
@@ -73,29 +94,55 @@ func (s *state) SetParticipants(addrs []mino.Address) {
 	s.Unlock()
 }
 
+func (s *state) GetPublicKeys() []kyber.Point {
+	s.Lock()
+	defer s.Unlock()
+	return s.publicKeys
+}
+
+func (s *state) SetPublicKeys(publicKeys []kyber.Point) {
+	s.Lock()
+	s.publicKeys = publicKeys
+	s.Unlock()
+}
+
+func (s *state) GetThreshold() int {
+	s.Lock()
+	defer s.Unlock()
+	return s.threshold
+}
+
+func (s *state) SetThreshold(threshold int) {
+	s.Lock()
+	s.threshold = threshold
+	s.Unlock()
+}
+
 // Handler represents the RPC executed on each node
 //
 // - implements mino.Handler
 type Handler struct {
 	mino.UnsupportedHandler
 	sync.RWMutex
-	dkg       *pedersen.DistKeyGenerator
-	privKey   kyber.Scalar
-	me        mino.Address
-	privShare *share.PriShare
-	startRes  *state
-	deals     chan dealFrom
-	responses chan responseFrom
+	dkg            *pedersen.DistKeyGenerator
+	privKey        kyber.Scalar
+	me             mino.Address
+	privShare      *share.PriShare
+	startRes       *state
+	deals          chan dealFrom
+	dealsResharing chan dealFromResharing
+	responses      chan responseFrom
 }
 
 // NewHandler creates a new handler
 func NewHandler(privKey kyber.Scalar, me mino.Address) *Handler {
 	return &Handler{
-		privKey:   privKey,
-		me:        me,
-		startRes:  &state{},
-		deals:     make(chan dealFrom, 100),
-		responses: make(chan responseFrom, 100),
+		privKey:        privKey,
+		me:             me,
+		startRes:       &state{},
+		deals:          make(chan dealFrom, 1000),
+		dealsResharing: make(chan dealFromResharing, 1000),
+		responses:      make(chan responseFrom, 1000),
```

Comment:
the initial `100` size was pretty arbitrary. Do you have a good reason to require a `1000`-length channel, or is that also arbitrary ?  😅

Reply:
because I needed to test the code for 128 nodes or more.